x/evm/vm/interface.go

```diff
@@ -0,0 +1,52 @@
+package vm
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/holiman/uint256"
+)
+
+// PrecompiledContracts defines a map of address -> precompiled contract
+type PrecompiledContracts map[common.Address]vm.PrecompiledContract
+
+type StatefulPrecompiledContract interface {
+	vm.PrecompiledContract
+	RunStateful(evm EVM, addr common.Address, input []byte, value *big.Int) (ret []byte, err error)
+}
+
+// EVM defines the interface for the Ethereum Virtual Machine used by the EVM module.
+type EVM interface {
```

Comment:
This seems very general a name. Could you mention a bit more on what this is supposed to do? And its implementation?

Reply:
'GethEVM' is what I'd suggest.

Reply:
this is just the EVM interface (common set of funcs). The Geth EVM is one example of a concrete implementation